player_test.py

```diff
@@ -251,23 +256,25 @@ def test_user_can_make_another_choice_if_typo_during_input(self):
   # test 44
     mario = Player("placeholder name")
     allowed_choices = ['front', 'back']
+    prompt_text = "please pick a direction to collect food from"
     expected_result = "front"
     input_patcher = mock.patch('builtins.input', side_effect = ["fromt", "front"])
     InputMock = input_patcher.start()
-    mario.choose_direction_to_pick_food(allowed_choices)
-    self.assertEqual(mario.user_choice_direction, expected_result)
+    mario.make_choice(allowed_choices, prompt_text)
+    self.assertEqual(mario.user_choice, expected_result)
     self.assertEqual(InputMock.call_count, 2)
     input_patcher.stop()
 
-  def test_addrsses_user_by_name_when_asking_for_direction_choice(self):
+  def test_addresses_user_by_name_when_asking_for_direction_choice(self):
```

Comment:
I can see why you've kept the ant and direction tests separate, because that's how you developed it because they were originally separate functions. However, the new `make_choice` function doesn't care whether the options are ants or directions, so generalising the tests would be a good idea. For example `test_the_first_string_from_a_list_of_one_string_is_a_valid_choice`, `test_the_first_string_from_a_list_of_several_strings_is_a_valid_choice`, `test_a_middle_string_from_a_list_of_several_strings_is_a_valid_choice`, `test_the_last_string_from_a_list_of_several_strings_is_a_valid_choice`.